lib/azureagent.js

```diff
@@ -0,0 +1,82 @@
+var https   = require('https');
+var debug   = require('debug')('azure-agent');
+var util    = require('util');
+
+/**
+ * Idle socket timeout, set to 55 seconds because the Azure load balancer will
+ * silently drop connections after 60 s of idle time. And we won't detect the
+ * drop until we try to send keep-alive packages, so to avoid making requests
+ * on invalid connections we close connections after 55 s of idle time.
+ */
+var SOCKET_IDLE_TIMEOUT = 55 * 1000;
+
+/**
+ * Error handler that ignores errors which occurs while the socket is owned
+ * AzureAgent. We don't want these errors to crash our application.
+ */
+var idleErrorHandler = function idleErrorHandler(err) {
```

Comment:
hrm js nit... if your using strict mode (you are by default in babel) then there is not much reason to both assign this with var and assign a name to the function (ideally you just do the latter).